src/sdl2/render.rs

```diff
@@ -252,24 +256,25 @@ impl<T> Deref for Canvas<T> {
 /// Its context may be shared with the `TextureCreator`.
 ///
 /// The context will not be dropped until all references of it are out of scope.
-pub struct Canvas<T> {
+pub struct Canvas<T, TC> {
     target: T,
-    context: Rc<RendererContext>,
+    context: Rc<RendererContext<TC>>,
 }
 
 /// Alias for a `Canvas` that was created out of a `Surface`
-pub type SurfaceCanvas<'s> = Canvas<Surface<'s>>;
+pub type SurfaceCanvas<'s> = Canvas<Surface<'s>, SurfaceContext<'s>>;
 
 /// Methods for the `SurfaceCanvas`.
-impl<'s> Canvas<Surface<'s>> {
+impl<'s> Canvas<Surface<'s>, SurfaceContext<'s>> {
```

Comment:
That was my first idea too but unfortunately that would not work with the TextureTarget.

The TextureTarget can have two different parent context's, a window or surface. Even though a Canvas with a TextureTarget cannot outlive the original Canvas, due to the lifetimes, it is still very important that it has the same parent context as the parent canvas, because we do not want to create a new context and have this context drop, because it will trigger a renderer destroy.